### PR TITLE
feat(logs): custom Role for Kinesis destination

### LINKED ISSUE #7661: logs: Allow overriding Role in `addSubscriptionFilter`
I'm trying to add a boundary permission to the IAM role created here:
https://github.com/aws/aws-cdk/blob/v1.36.0/packages/@aws-cdk/aws-logs-destinations/lib/kinesis.ts#L17

My code looks something like:
```
item.addSubscriptionFilter(generateID(),{destination: dest, filterPattern: log.FilterPattern.allEvents()} )
```
This creates an IAM role in the background.

It looks like you should be able to override the role like with the id:
```
        const id = 'CloudWatchLogsCanPutRecords';
        new iam.Role(this, id, {
            assumedBy: new iam.ServicePrincipal('logs.amazonaws.com'),
            permissionsBoundary: boundary
        });
```
But the scope (or `this`) used by `addSubscriptionFilter` is actually inaccessible.
https://github.com/aws/aws-cdk/blob/47c991949618d0e95bfafccb0ecd73687036b432/packages/%40aws-cdk/aws-logs/lib/log-group.ts#L107

### Use Case
To add boundary permission to IAM role

### Proposed Solution

Allow optional iam role to be passed in with `addSubscriptionFilter`

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by gmeligio**
* [x] Testing
  - Unit test added.
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated

When creating a logs destination, an optional `role` can be passed to be assumed when writing logs in the destination.
 
Closes #7661.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by gmeligio** [comment]
I was able to unit test `aws-logs` but no `aws-logs-destinations` because the latter has peer-dependencies that are not available with `yarn install` like `@aws-cdk/aws-iam` and `@aws-cdk/aws-lambda`.

File: packages/@aws-cdk/aws-logs/README.md [hunk 1]
```diff
@@ -93,6 +93,17 @@ new SubscriptionFilter(this, 'Subscription', {
...
 ```
 
+You can pass an existing IAM ``roleArn`` to be assumed for writing logs in the destination. If not, a new one will be created.
+
+```
+new SubscriptionFilter(this, 'Subscription', {
+    logGroup,
+    destination: new LogsDestinations.LambdaDestination(fn),
+    filterPattern: FilterPattern.allTerms("ERROR", "MainThread")
+    roleArn: role.roleArn
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-logs/README.md:103]: It's not roleArn though, right?

File: packages/@aws-cdk/aws-logs/lib/log-group.ts [hunk 2]
```diff
@@ -23,6 +23,14 @@ export interface ILogGroup extends IResource {
    */
   readonly logGroupName: string;
 
+  /**
+   * The role to assume to write log events to the destination
+   *
+   * @default No role assumed
+   * @attribute
+   */
+  readonly role?: iam.IRole;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-logs/lib/log-group.ts:32]: No need to expose this here.

**Review by rix0rrr — CHANGES_REQUESTED** [review]
It looks like what you're trying to do is preventing the Kinesis destination from creating a Role for you. I'm assuming because your organization does not allow the creation of new Roles by developers.

Right now it's the case that the `ILogSubscriptionDestination` is responsible for creating the Roles. If you want it to stop doing that, the easiest way is to make a pre-existing Role configurable on all the individual subclasses of `ILogSubscriptionDestination`, starting with the Kinesis one. Create a new (optional) Props object with a single `role?: Role` field, and use that one if given instead of creating a new one. This is the pattern used all across the CDK.

----

The next question might be--does this make sense? Should the Role not be configured by the subscription filter itself instead of its destination?

A case could definitely be made for that. That requires a bigger refactor, where we add the `role?: Role` argument to `SubscriptionFilterOptions`, and we need to add some protocol between the SubscriptionFilter and the Destination to indicate whether or not the filter already HAS a role or needs one configured (and even if one already exists, permissions may still need to be added to them!).

You could also take on that, I'm not opposed to it.

-----

In neither case is a change to `LogGroup` necessary, or does `LogGroup` implicitly get a Role or something. You are using the LogGroup object to transport a single variable somewhere else -- that doesn't seem necessargy.

**Review by rix0rrr — APPROVED** [review]

**Review by rix0rrr — APPROVED** [review]

**Review by rix0rrr — DISMISSED** [review]

**Review by rix0rrr — DISMISSED** [review]

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**